zxlive/proof_actions.py

```diff
@@ -112,17 +115,90 @@ def rewriter() -> None:
             return rewriter
         for action in self.actions:
             if action.button is not None: continue
-            btn = QPushButton(action.name,parent)
+            btn = QPushButton(action.name, parent)
             btn.setMaximumWidth(150)
             btn.setStatusTip(action.tooltip)
             btn.setEnabled(False)
-            btn.clicked.connect(create_rewrite(action,parent))
+            btn.clicked.connect(create_rewrite(action, parent))
             self.btn_group.addButton(btn)
             action.button = btn
 
     def update_active(self, g: GraphT, verts: List[VT], edges: List[ET]) -> None:
         for action in self.actions:
-            action.update_active(g,verts,edges)
+            action.update_active(g, verts, edges)
+
+
+def to_networkx(graph: Graph) -> nx.Graph:
```

Comment:
Using networkx is for now I think fine, but when we want to do matching where we also have variables and parameters, then we might need to roll our own implementation.

Reply:
The networkx implementation for matching graphs looks quite involved. It would be better to not reimplement it if it is not necessary. The networkx graph matcher takes in a semantic_feasibility function as an optional argument. It is worth checking if passing a semantic_feasibility function is enough to deal with variables and parameters.

Reply:
It could be that the networkx implementation is involved because it is doing a clever graph iso algorithm. Basic iso checking via depth-first traversal is not hard to implement, and would probably be fast enough.

Reply:
You're right, it implements the VF2++ algorithm (https://doi.org/10.1016/j.dam.2018.02.018). We can do our own implementation when we start looking at variables and parameters, but let's do that after the [poly](https://github.com/Quantomatic/zxlive/tree/poly) branch is merged.